include/crm/common/xml.h

```diff
@@ -329,7 +329,7 @@ int xml_apply_patchset(xmlNode *xml, xmlNode *patchset, bool check_version);
 void patchset_process_digest(xmlNode *patch, xmlNode *source, xmlNode *target, bool with_digest);
 
 void save_xml_to_file(xmlNode * xml, const char *desc, const char *filename);
-char *xml_get_path(xmlNode *xml);
+char *xml_get_path(const xmlNode *xml);
```

Comment:
FYI, this is public API due to being in a public header. Normally that means we can't change the interface without breaking backward compatibility, which requires a shared library version bump that requires all client applications using the library to be rebuilt. Thus we only do that on very rare occasions (around every 3-6 years -- keep in mind that most distros don't like incompatible library changes within one of their major releases).

However we're lucky here. Since a non-const pointer can be used where a const pointer is required, adding const to a function argument does not break any code linked against the former version, making it a compatible change. The reverse is not true; removing const from a pointer argument would break any client code passing a const pointer.

Reply:
Good to know. `xml_get_path()` and `pcmk__element_xpath()` don't modify the XML, so normally I'd use a const for that kind of thing. I noticed that a lot of our const-looking XML parsing functions don't use the const specifier though.

Come to think of it, IIRC I tried a `crm_log_xml_trace()` on a `const xmlNode` from those modified functions last night and it failed because `log_data_element()` discards the const qualifier. So, maybe a bad idea. Allowing tracing on a const would require adding more consts further down the line if it would be possible at all.

Reply:
Yep, that's often an issue, whether other pacemaker functions or libxml2 functions that don't specify const when they could.

FYI anything with a double underbar is internal API whose interface can be changed at will. Not all internal functions have been renamed like that yet but we're getting there.